lexibank_chenhmongmien.py

```diff
@@ -34,90 +33,63 @@ class Dataset(NonSplittingDataset):
     concept_class = HConcept
     language_class = HLanguage
 
-    def cmd_download(self, **kw):
-        wp = requests.get(
-            "https://en.wiktionary.org/wiki/Appendix:Hmong-Mien_comparative_vocabulary_list"
-        )
-        soup = BeautifulSoup(wp.content, "html.parser")
-
-        language_table_header, language_table = [], []
-        languages = soup.findAll("table", {"class": "wikitable sortable"})[0]
-        for lh in languages.findAll("th"):
-            language_table_header.append(lh.get_text().rstrip("\n"))
-
-        for r in languages.findAll("tr"):
-            temp = []
-            for cell in r.findAll("td"):
-                temp.append(cell.get_text().rstrip("\n"))
-            language_table.append(temp)
-
-        language_table = [x for x in language_table if x != []]
+    def cmd_download(self, args):
+        with self.raw_dir.temp_download(
+            "https://en.wiktionary.org/wiki/Appendix:Hmong-Mien_comparative_vocabulary_list",
+            'raw.html'
+        ) as p:
+            soup = BeautifulSoup(p.read_text(encoding='utf8'), "html.parser")
 
-        vob_table_header, vob_table = [], []
-        vob = soup.findAll("table", {"class": "wikitable sortable"})[1]
-        for vh in vob.findAll("th"):
-            vob_table_header.append(vh.get_text().rstrip("\n"))
+        def iter_rows(table):
+            yield [c.get_text().rstrip('\n') for c in table.findAll('th')]
+            for row in table.findAll('tr'):
+                yield [c.get_text().rstrip('\n') for c in row.findAll('td')]
 
-        for v in vob.findAll("tr"):
-            vtemp = []
-            for vcell in v.findAll("td"):
-                vtemp.append(vcell.get_text().rstrip("\n"))
-            vob_table.append(vtemp)
+        self.raw_dir.write_csv(
```

Comment:
`DataDir` can also help with CSV writing!

Reply:
@MacyL, please cheeck this line